sklego/meta.py

```diff
@@ -54,6 +54,47 @@ def __init__(self, estimator, groups, use_fallback=True):
         self.groups = groups
         self.use_fallback = use_fallback
 
+    def __check_group_cols_exist(self, X):
+        try:
+            x_cols = set(X.columns)
+        except AttributeError:
+            try:
+                ncols = X.shape[1]
+            except IndexError:
+                ncols = 1
+            x_cols = set(range(ncols))
+
+        # Check whether grouping columns exist
+        diff = set(as_list(self.groups)) - x_cols
+
+        if len(diff) > 0:
+            raise KeyError(f'{diff} not in columns of X ({x_cols})')
+
+    @staticmethod
+    def __check_missing_and_inf(X):
+        """Check that all elements of X are non-missing and finite"""
+        if np.any(pd.isnull(X)):
+            raise ValueError("X has NaN values")
+        try:
```

Comment:
i wonder ... why do we need a try/except here?

Reply:
The check for infinities is only relevant in case of numeric variables. If the check raises `TypeErrror`, the values are not numeric (and cannot be converted) and so checking for infinities does not make sense. Moved the comment at the exception to the try for more clarity.